### PR TITLE
Feature/154 request class

### PR DESCRIPTION
**Pull request by theblockstalk**
Upgrades

- new VerifiableCredential and VerifiableCredentialWithType class in `./src/util/ssi/vc.ts`
- implemented these `./src/util/request.ts` to have full Typescript report
- implemented these in `./src/services/communication` (changed Communication and Messages classes) to have full Typescript support
- any data sent by URL is doing using `?payload={base64Urlpayload}` structure and base64url type
- `user.getUsername()` is used everywhere and always returns a `TonomyUsername` type
- updated all tests and SDK to use the above

### CODE REVIEW

File: src/sdk/util/ssi/vc.ts [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+        if (typeof vc === 'string') {
+            this.vc = new VerifiableCredential<{ payload: T; type: string }>(vc);
+        } else if (vc instanceof VerifiableCredentialWithType) {
+            this.vc = vc.getVc() as VerifiableCredential<{ payload: T; type: string }>;
+        } else {
+            this.vc = vc;
+        }
+
+        this.decodedPayload = this.getVc().getCredentialSubject().payload;
+    }
```
Comment (Rebal67) [inline src/sdk/util/ssi/vc.ts:254]: why creating a new class for this cant be an optional property ?

File: src/sdk/util/time.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export function toDateTime(secs: number) {
+    const t = new Date(1970, 0, 1); // Epoch
+
+    t.setSeconds(secs);
+    return t;
+}
```
Comment (Rebal67) [inline src/sdk/util/time.ts:6]: function not clear

File: src/sdk/util/username.ts [hunk 3]
```diff
@@ -69,7 +71,28 @@ export class TonomyUsername {
...
+
+    /**
+     * Used in JSON.stringify
+     * @returns the username
+     *
+     * @throws Error if username is not set
+     */
+    toJSON(): string {
+        return this.toString();
+    }
```
Comment (Rebal67) [inline src/sdk/util/username.ts:97]: naming and comments make this unclear

File: src/sdk/util/ssi/vc.ts (hunk 1 above)
Comment (theblockstalk) [reply src/sdk/util/ssi/vc.ts:254]: talked with Rebal and accepted

**Pull request merged**